### ISSUE TITLE
Zim export doesn't work on headless server

### ISSUE THREAD

**Issue by shivams** [zim-desktop-wiki/zim-desktop-wiki#579]
I tried running the Zim's export command inside a Vagrant machine which has no GUI interface. It didn't show any error for the notebook, but it also didn't produce any output.

The OS inside this machine is Ubuntu 16.04.

The command that I used to export the Zim notebook to HTML is this:

```shell
zim --export --output=/home/ubuntu/html --format=html ./ 
```

**Comment by jaap-karssenberg** [comment]
Was Gtk installed on this machine, or also not?

**Comment by shivams** [comment]
It was not installed. However, I just installed it with `sudo apt install python-gtk2-dev`, but even then the export is not working.

**Comment by jaap-karssenberg** [comment]
Never tested zim without gtk import - in theory it should work, but
probably needs some polishing of the module imports.

-- Jaap

**Comment by edrex** [comment]
I'd like to get this working. 

Today I set up systemd units to auto export and publish to netlify whenever there's a change:

**Comment by jaap-karssenberg** [comment]
Probably it is not hard, just fixing a few imports.

The export code itself and the underlying notebook data structures don't
even require Gtk.

If you have the error logs (preferable based on zim version 0.70) should be
straightforward to identify the bottleneck.

-- Jaap

**Comment by MeisterP** [comment]
This is on my normal desktop machine. Running `unset DISPLAY` before running `zim --export`. 
```
(zim:10859): Gdk-CRITICAL **: 07:37:02.963: gdk_keymap_get_for_display: assertion 'GDK_IS_DISPLAY (display)' failed

(zim:10859): Gdk-CRITICAL **: 07:37:02.963: gdk_keymap_get_modifier_mask: assertion 'GDK_IS_KEYMAP (keymap)' failed

(zim:10859): Gdk-CRITICAL **: 07:37:02.963: gdk_keymap_get_for_display: assertion 'GDK_IS_DISPLAY (display)' failed

(zim:10859): Gtk-CRITICAL **: 07:37:02.963: _gtk_replace_virtual_modifiers: assertion 'GDK_IS_KEYMAP (keymap)' failed

(zim:10859): Gtk-CRITICAL **: 07:37:02.984: gtk_icon_theme_get_for_screen: assertion 'GDK_IS_SCREEN (screen)' failed

(zim:10859): Gtk-CRITICAL **: 07:37:02.985: gtk_icon_theme_get_for_screen: assertion 'GDK_IS_SCREEN (screen)' failed

(zim:10859): Gtk-CRITICAL **: 07:37:02.985: gtk_icon_theme_get_for_screen: assertion 'GDK_IS_SCREEN (screen)' failed

(zim:10859): Gtk-CRITICAL **: 07:37:02.985: gtk_icon_theme_get_for_screen: assertion 'GDK_IS_SCREEN (screen)' failed

(zim:10859): Gtk-CRITICAL **: 07:37:02.985: gtk_icon_theme_get_for_screen: assertion 'GDK_IS_SCREEN (screen)' failed

(zim:10859): Gtk-CRITICAL **: 07:37:02.986: gtk_icon_theme_get_for_screen: assertion 'GDK_IS_SCREEN (screen)' failed

(zim:10859): Gtk-CRITICAL **: 07:37:02.986: gtk_icon_theme_get_for_screen: assertion 'GDK_IS_SCREEN (screen)' failed

(zim:10859): Gtk-CRITICAL **: 07:37:02.986: gtk_icon_theme_get_for_screen: assertion 'GDK_IS_SCREEN (screen)' failed

(zim:10859): Gtk-CRITICAL **: 07:37:02.986: gtk_icon_theme_get_for_screen: assertion 'GDK_IS_SCREEN (screen)' failed

(zim:10859): Gtk-CRITICAL **: 07:37:02.986: gtk_icon_theme_get_for_screen: assertion 'GDK_IS_SCREEN (screen)' failed

(zim:10859): Gtk-CRITICAL **: 07:37:02.986: gtk_icon_theme_get_for_screen: assertion 'GDK_IS_SCREEN (screen)' failed

(zim:10859): Gtk-CRITICAL **: 07:37:03.009: gtk_clipboard_get_for_display: assertion 'display != NULL' failed

(zim:10859): Gtk-CRITICAL **: 07:37:03.009: gtk_clipboard_get_for_display: assertion 'display != NULL' failed
ERROR: Exception while loading plugin: insertsymbol
Traceback (most recent call last):
  File "/usr/lib64/python3.6/site-packages/zim/plugins/__init__.py", line 360, in load_plugins_from_preferences
    self.load_plugin(name)
  File "/usr/lib64/python3.6/site-packages/zim/plugins/__init__.py", line 461, in load_plugin
    klass = self.get_plugin_class(name)
  File "/usr/lib64/python3.6/site-packages/zim/plugins/__init__.py", line 415, in get_plugin_class
    mod = get_module(modname)
  File "/usr/lib64/python3.6/site-packages/zim/utils.py", line 35, in get_module
    mod = __import__(name)
  File "/usr/lib64/python3.6/site-packages/zim/plugins/insertsymbol.py", line 11, in <module>
    from zim.gui.pageview import PageViewExtension
  File "/usr/lib64/python3.6/site-packages/zim/gui/pageview.py", line 3471, in <module>
    CURSOR_TEXT = Gdk.Cursor.new_from_name(Gdk.Display.get_default(), 'text')
TypeError: Argument 0 does not allow None as a value
ERROR: Exception while loading plugin: versioncontrol
Traceback (most recent call last):
  File "/usr/lib64/python3.6/site-packages/zim/plugins/__init__.py", line 360, in load_plugins_from_preferences
    self.load_plugin(name)
  File "/usr/lib64/python3.6/site-packages/zim/plugins/__init__.py", line 461, in load_plugin
    klass = self.get_plugin_class(name)
  File "/usr/lib64/python3.6/site-packages/zim/plugins/__init__.py", line 415, in get_plugin_class
    mod = get_module(modname)
  File "/usr/lib64/python3.6/site-packages/zim/utils.py", line 35, in get_module
    mod = __import__(name)
  File "/usr/lib64/python3.6/site-packages/zim/plugins/versioncontrol/__init__.py", line 29, in <module>
    from zim.gui.mainwindow import MainWindowExtension
  File "/usr/lib64/python3.6/site-packages/zim/gui/mainwindow.py", line 34, in <module>
    from zim.gui.pageview import PageView
  File "/usr/lib64/python3.6/site-packages/zim/gui/pageview.py", line 3471, in <module>
    CURSOR_TEXT = Gdk.Cursor.new_from_name(Gdk.Display.get_default(), 'text')
TypeError: Argument 0 does not allow None as a value
ERROR: Exception while loading plugin: attachmentbrowser
Traceback (most recent call last):
  File "/usr/lib64/python3.6/site-packages/zim/plugins/__init__.py", line 360, in load_plugins_from_preferences
    self.load_plugin(name)
  File "/usr/lib64/python3.6/site-packages/zim/plugins/__init__.py", line 461, in load_plugin
    klass = self.get_plugin_class(name)
  File "/usr/lib64/python3.6/site-packages/zim/plugins/__init__.py", line 415, in get_plugin_class
    mod = get_module(modname)
  File "/usr/lib64/python3.6/site-packages/zim/utils.py", line 35, in get_module
    mod = __import__(name)
  File "/usr/lib64/python3.6/site-packages/zim/plugins/attachmentbrowser/__init__.py", line 47, in <module>
    from zim.gui.pageview import PageViewExtension
  File "/usr/lib64/python3.6/site-packages/zim/gui/pageview.py", line 3471, in <module>
    CURSOR_TEXT = Gdk.Cursor.new_from_name(Gdk.Display.get_default(), 'text')
TypeError: Argument 0 does not allow None as a value
ERROR: Exception while loading plugin: spell
Traceback (most recent call last):
  File "/usr/lib64/python3.6/site-packages/zim/plugins/__init__.py", line 360, in load_plugins_from_preferences
    self.load_plugin(name)
  File "/usr/lib64/python3.6/site-packages/zim/plugins/__init__.py", line 461, in load_plugin
    klass = self.get_plugin_class(name)
  File "/usr/lib64/python3.6/site-packages/zim/plugins/__init__.py", line 415, in get_plugin_class
    mod = get_module(modname)
  File "/usr/lib64/python3.6/site-packages/zim/utils.py", line 35, in get_module
    mod = __import__(name)
  File "/usr/lib64/python3.6/site-packages/zim/plugins/spell.py", line 17, in <module>
    from zim.gui.pageview import PageViewExtension
  File "/usr/lib64/python3.6/site-packages/zim/gui/pageview.py", line 3471, in <module>
    CURSOR_TEXT = Gdk.Cursor.new_from_name(Gdk.Display.get_default(), 'text')
TypeError: Argument 0 does not allow None as a value
ERROR: Exception while loading plugin: pageindex
Traceback (most recent call last):
  File "/usr/lib64/python3.6/site-packages/zim/plugins/__init__.py", line 360, in load_plugins_from_preferences
    self.load_plugin(name)
  File "/usr/lib64/python3.6/site-packages/zim/plugins/__init__.py", line 461, in load_plugin
    klass = self.get_plugin_class(name)
  File "/usr/lib64/python3.6/site-packages/zim/plugins/__init__.py", line 415, in get_plugin_class
    mod = get_module(modname)
  File "/usr/lib64/python3.6/site-packages/zim/utils.py", line 35, in get_module
    mod = __import__(name)
  File "/usr/lib64/python3.6/site-packages/zim/plugins/pageindex/__init__.py", line 21, in <module>
    from zim.gui.pageview import PageViewExtension
  File "/usr/lib64/python3.6/site-packages/zim/gui/pageview.py", line 3471, in <module>
    CURSOR_TEXT = Gdk.Cursor.new_from_name(Gdk.Display.get_default(), 'text')
TypeError: Argument 0 does not allow None as a value
ERROR: Exception while loading plugin: linesorter
Traceback (most recent call last):
  File "/usr/lib64/python3.6/site-packages/zim/plugins/__init__.py", line 360, in load_plugins_from_preferences
    self.load_plugin(name)
  File "/usr/lib64/python3.6/site-packages/zim/plugins/__init__.py", line 461, in load_plugin
    klass = self.get_plugin_class(name)
  File "/usr/lib64/python3.6/site-packages/zim/plugins/__init__.py", line 415, in get_plugin_class
    mod = get_module(modname)
  File "/usr/lib64/python3.6/site-packages/zim/utils.py", line 35, in get_module
    mod = __import__(name)
  File "/usr/lib64/python3.6/site-packages/zim/plugins/linesorter.py", line 14, in <module>
    from zim.gui.pageview import PageViewExtension
  File "/usr/lib64/python3.6/site-packages/zim/gui/pageview.py", line 3471, in <module>
    CURSOR_TEXT = Gdk.Cursor.new_from_name(Gdk.Display.get_default(), 'text')
TypeError: Argument 0 does not allow None as a value
ERROR: Exception while loading plugin: tags
Traceback (most recent call last):
  File "/usr/lib64/python3.6/site-packages/zim/plugins/__init__.py", line 360, in load_plugins_from_preferences
    self.load_plugin(name)
  File "/usr/lib64/python3.6/site-packages/zim/plugins/__init__.py", line 461, in load_plugin
    klass = self.get_plugin_class(name)
  File "/usr/lib64/python3.6/site-packages/zim/plugins/__init__.py", line 415, in get_plugin_class
    mod = get_module(modname)
  File "/usr/lib64/python3.6/site-packages/zim/utils.py", line 35, in get_module
    mod = __import__(name)
  File "/usr/lib64/python3.6/site-packages/zim/plugins/tags.py", line 16, in <module>
    from zim.plugins.pageindex import PageTreeStore, PageTreeStoreBase, PageTreeView, \
  File "/usr/lib64/python3.6/site-packages/zim/plugins/pageindex/__init__.py", line 21, in <module>
    from zim.gui.pageview import PageViewExtension
  File "/usr/lib64/python3.6/site-packages/zim/gui/pageview.py", line 3471, in <module>
    CURSOR_TEXT = Gdk.Cursor.new_from_name(Gdk.Display.get_default(), 'text')
TypeError: Argument 0 does not allow None as a value
```

**Comment by shivams** [comment]
I just saw this old issue. As of now, the zim export does seem to be working fine. Today, I tried it with a Vagrant machine of Ubuntu 18.04 and installed zim in it and tried importing to HTML, and it was working fine. I do not know what changed (except that I am using 18.04 instead of 16.04 VM).

Further, I also came across a Gitlab CI sample config for serving static website through Gitlab here: https://gitlab.com/pages/zim/. The CI config here uses shared runners (public docker machines) to import the notebook to html using Zim command-line and then serve them over the web. I tested it and it is also working fine.

So, I think the issue can safely be closed for now.
